src/doc/src/reference/resolver.md

```diff
@@ -7,8 +7,10 @@ result of the resolution is stored in the `Cargo.lock` file which "locks" the
 dependencies to specific versions, and keeps them fixed over time.
 
 The resolver attempts to unify common dependencies while considering possibly
-conflicting requirements. The sections below provide some details on how these
-constraints are handled, and how to work with the resolver.
+conflicting requirements. Dependency resolution is, however, an NP-hard problem,
+and so the resolver uses a heuristic algorithm that may yield counterintuitive
+results in some cases. The sections below provide some details on how
```

Comment:
I would be reluctant to start the chapter emphasizing that Cargo's primary job may be "counterintuitive". I personally would prefer to leave out this new sentence. Can you say what the intent of it is?

Reply:
I think that someone new to dependency management might naively think, "OK, Cargo gets all the requirements, finds the versions that work, easy enough" — I don't think it's obvious to a newcomer that dependency-resolution is, indeed, an NP-hard problem where there may be no clear "best" solution to converge to. My intent here was to raise that idea — "this might sound straightforward but actually it isn't always!"